src/Source.cpp

```diff
@@ -29,19 +30,18 @@ ProcessMessageResult Source::process_message(FlatbufferMessage const &Message) {
       LOG(Sev::Debug, "!_hdf_writer_module for {}", SourceName);
       return ProcessMessageResult::ERR;
     }
-    auto ret = WriterModule->write(Message);
-    _cnt_msg_written += 1;
-    _processed_messages_count += 1;
-    if (ret.is_ERR()) {
-      if (log_level >= static_cast<int>(Sev::Debug)) {
-        LOG(Sev::Debug, "Failure while writing message: {}", ret.to_str());
+    try {
+      WriterModule->write(Message);
+      _cnt_msg_written += 1;
+      _processed_messages_count += 1;
+      if (HDFFileForSWMR != nullptr) {
+        HDFFileForSWMR->SWMRFlush();
       }
+      return ProcessMessageResult::OK;
+    } catch (const HDFWriterModuleRegistry::WriterException &E) {
+      LOG(Sev::Debug, "Failure while writing message: {}", E.what());
```

Comment:
I think this should be `Sev::Error`. Data not ending up in the file is v bad.
Looks like that was the behaviour before: https://github.com/ess-dmsc/kafka-to-nexus/pull/405/files#diff-b5e422a9fabe9330754618aaf296930eL45